File: src/main/java/org/goafabric/core/data/logic/OrganizationLogic.java
package org.goafabric.core.data.logic;

import jakarta.enterprise.context.ApplicationScoped;
import jakarta.transaction.Transactional;
import org.goafabric.core.data.controller.vo.Organization;
import org.goafabric.core.data.logic.mapper.OrganizationMapper;
import org.goafabric.core.data.repository.OrganizationRepository;

import java.util.List;

@ApplicationScoped
@Transactional
public class OrganizationLogic {
    private final OrganizationMapper organizationMapper;

    private final OrganizationRepository organizationRepository;

    public OrganizationLogic(OrganizationMapper organizationMapper, OrganizationRepository organizationRepository) {
        this.organizationMapper = organizationMapper;
        this.organizationRepository = organizationRepository;
    }

    public Organization getById(String id) {
        return organizationMapper.map(
                organizationRepository.findById(id));
    }

    public void deleteById(String id) {
        organizationRepository.deleteById(id);
    }

    public List<Organization> findByName(String name) {
        return organizationMapper.map(
                organizationRepository.findByNameStartsWithIgnoreCase(name));
    }

    public Organization save(Organization organization) {
        return organizationMapper.map(organizationRepository.save(
                organizationMapper.map(organization)));
    }


}

File: src/main/kotlin/org/goafabric/core/organization/logic/OrganizationLogic.kt
package org.goafabric.core.organization.logic

import jakarta.enterprise.context.ApplicationScoped
import jakarta.transaction.Transactional
import org.goafabric.core.organization.controller.dto.Organization
import org.goafabric.core.organization.logic.mapper.OrganizationMapper
import org.goafabric.core.organization.persistence.OrganizationRepository

@ApplicationScoped
@Transactional
class OrganizationLogic(
    private val mapper: OrganizationMapper,
    private val repository: OrganizationRepository
) {
    fun getById(id: String): Organization = mapper.map(repository.findById(id) ?: throw IllegalArgumentException("Organization not found: $id"))

    fun deleteById(id: String) = repository.deleteById(id)

    fun findByName(name: String): List<Organization> =
        mapper.map(repository.findByNameStartsWith(name))

    fun save(organization: Organization): Organization =
        mapper.map(repository.save(mapper.map(organization)))
}
